### PR TITLE
iron-functions 0.2.68 (new formula)

### PR DESCRIPTION
**Pull request by merqlove**
Add [Iron.io](https://www.iron.io) `functions` CLI tools.
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/functions.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    dir = buildpath/"src/github.com/iron-io/functions"
+    dir.install Dir["*"]
+    cd dir/"fn" do
+      system "make", "dep"
+      system "go", "build", "-o", bin/"fn"
+    end
+  end
+
+  test do
+    system bin/"fn", "-help"
```
Comment (fxcoudert) [inline Formula/functions.rb]: The test should exercise the tool's functionality. Perform an operation, match the output or the files created.

File: Formula/functions.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  depends_on "go" => :build
+  depends_on "dep" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    dir = buildpath/"src/github.com/iron-io/functions"
+    dir.install Dir["*"]
+    cd dir/"fn" do
+      system "make", "dep"
+      system "go", "build", "-o", bin/"fn"
```
Comment (fxcoudert) [inline Formula/functions.rb]: Is there no other file to install? Man page, doc, config file, etc?

**Review by fxcoudert — CHANGES_REQUESTED** [review]
Commit message should be `functions 0.2.62 (new formula)`

**Comment by ilovezfs** [comment]
This will also need a less generic name.

File: Formula/functions.rb (hunk 2 above)
Comment (merqlove) [reply Formula/functions.rb]: No

File: Formula/functions.rb (hunk 1 above)
Comment (merqlove) [reply Formula/functions.rb]: I can add match version

Comment (ilovezfs) [reply Formula/functions.rb]: It actually needs to test the software is functional. See `jose` and `faas-cli` for some examples.

**Comment by merqlove** [comment]
Ok, i’ll look on this week.
--
Best regards,
Alexander Merkulov

E-mail: sasha@merqlove.ru
Tel.:  +7(925) 278-44-98
Skype:  merqlove

**Comment by merqlove** [comment]
@fxcoudert, fixed test

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by merqlove** [comment]
@ilovezfs if this one is ok i'll change its name.

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
```
Error: 2 problems in 1 formula
functions:
  * C: 5: col 3: `url` (line 5) should be put before `version` (line 4)
  * Stable: version 0.2.62 is redundant with version scanned from URL
```

```
 Testing functions
/usr/bin/sandbox-exec -f /tmp/homebrew20171007-14242-19unq6.sb /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/bin/ruby -W0 -I /usr/local/Homebrew/Library/Homebrew -- /usr/local/Homebrew/Library/Homebrew/test.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/functions.rb --verbose
==> /usr/local/Cellar/functions/0.2.62/bin/fn -v 2>&1
Error: functions: failed
<1> expected but was
<0>.
```

File: Formula/functions.rb (hunk 2 above)
Comment (ilovezfs) [reply Formula/functions.rb]: should add `prefix.install_metafiles`

File: Formula/functions.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    end
+  end
+
+  test do
+    output = shell_output("#{bin}/fn -v 2>&1", 1)
```
Comment (ilovezfs) [inline Formula/functions.rb]: don't need this test since we have the other test below

File: Formula/functions.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    assert_equal version, output.chomp
+    output = shell_output("#{bin}/fn init --runtime go user/some 2>&1", 1)
+    assert_equal "runtime: go\nfunc.yaml created.", output.chomp
```
Comment (ilovezfs) [inline Formula/functions.rb]: Please use a heredoc to avoid `\n`

File: Formula/functions.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
+class Functions < Formula
+  desc "Go version of the IronFunctions command-line tools"
+  homepage "https://github.com/iron-io/functions"
+  version "0.2.62"
+  url "https://github.com/iron-io/functions/archive/#{version}.tar.gz"
+  sha256 "e39e67195afc61b4bce69d61aa1248be3e5980fa5adce432315021d98a093c7a"
+
+  depends_on "go" => :build
+  depends_on "dep" => :build
```
Comment (ilovezfs) [inline Formula/functions.rb]: alphabetize the build deps

File: Formula/functions.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
+class Functions < Formula
+  desc "Go version of the IronFunctions command-line tools"
+  homepage "https://github.com/iron-io/functions"
+  version "0.2.62"
```
Comment (ilovezfs) [inline Formula/functions.rb]: just inline it in the url please

**Review by ilovezfs — CHANGES_REQUESTED** [review]

**Comment by merqlove** [comment]
@BrewTestBot test this please

**Comment by merqlove** [comment]
Updated

**Comment by ilovezfs** [comment]
@merqlove how are we doing on a less generic name?

File: Formula/functions.rb [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    cd dir/"fn" do
+      system "make", "dep"
+      system "go", "build", "-o", bin/"fn"
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    output = shell_output("#{bin}/fn init --runtime go user/some 2>&1", 1)
+    assert_equal "runtime: go\nfunc.yaml created.", output.chomp
```
Comment (ilovezfs) [inline Formula/functions.rb]: should use a heredoc instead

Comment (merqlove) [reply Formula/functions.rb]: fixed

**Comment by merqlove** [comment]
@ilovezfs i will fix its name

**Comment by merqlove** [comment]
@ilovezfs name fixed

**Comment by merqlove** [comment]
@ilovezfs seems its ok now. i need squash commits?

**Comment by dunn** [comment]
Yes, please squash it to one commit!

**Comment by merqlove** [comment]
@dunn, @ilovezfs done

**Comment by dunn** [comment]
Thank you! Merged as ec6c6c29b2a3eff443eb6aeac6fa768fa05595a5.

**Pull request closed without merging**